### PR TITLE
AdminHomePageUiTest failing due to ajax-loader gif #3030

### LINKED ISSUE #3030: AdminHomePageUiTest failing due to ajax-loader gif
The test doesn't pass even if run again multiple times.

Maybe we can add a wait while testing or ignore these in verifyHtml?

image

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #3030

### CODE REVIEW

File: src/test/resources/pages/newlyJoinedInstructorFeedbacksPage.html [hunk 1]
```diff
@@ -460,7 +460,7 @@ <h5 class="col-sm-2">
                         Open
                     </span>
                 </td>
-                <td class="session-response-for-test recent"><img src="/images/ajax-loader.gif"></td>
+                <td class="session-response-for-test recent">0 / 9</td>
```
Comment (damithc) [inline src/test/resources/pages/newlyJoinedInstructorFeedbacksPage.html]: Just change this to {*} instead?

File: src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java [hunk 2]
```diff
@@ -144,7 +145,8 @@ private void testCreateInstructorAction() throws Exception {
         //verify sample course is accessible for newly joined instructor as an instructor
         
         ______TS("new instructor can see sample course in homepage");
-        InstructorHomePage instructorHomePage = AppPage.getNewPageInstance(browser, InstructorHomePage.class);  
+        InstructorHomePage instructorHomePage = AppPage.getNewPageInstance(browser, InstructorHomePage.class);
+        ThreadHelper.waitFor(2000);
```
Comment (damithc) [inline src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java]: We prefer not to add unconditional waits unnecessarily. This results in added 6 seconds for every test run for every developer from this point onwards, the cumulative time lost is not trivial.

Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java]: An alternative will be placing {*} in the place where the (0 / 9) is expected. Will that be better?

**Comment by LowWeiLin** [comment]
A more robust way is to wait for the ajax-loader gif to disappear. This caters to both slower and faster machines. Selenium provides an API to do this.

File: src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java (hunk 2 above)
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/AdminHomePageUiTest.java]: On second thought, I disagree. This is a meaningful comparison.

**Comment by wkurniawan07** [comment]
I have to give credit to whoever wrote that waitForElementToDisappear. Easy to extend to this ajax-loader.

**Comment by LowWeiLin** [comment]
See if it is possible/good to add this inside verifyHtmlMainContent() itself, similar to verifyHtmlAjax(). This way, the tests themselves don't have to have the additional line of code before calling verifyHtmlMainContent(). It will also fix other places with the same problem.

**Comment by wkurniawan07** [comment]
I'd want to say it's possible, but I can't say until I try and get dev green. Let's hope for the best.

File: src/test/resources/pages/instructorFeedbackDoneEditing.html [hunk 3]
```diff
@@ -751,7 +751,7 @@ <h5 class="col-sm-2">
                      </span>
                   </td>
                   <td class="session-response-for-test recent">
-                     <img src="/images/ajax-loader.gif"/>
+                     0 / 4
                                        </td>
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorFeedbackDoneEditing.html]: not actually part of the issue, but this needs to be fixed to ensure dev green. can this be included?

**Comment by wkurniawan07** [comment]
Works wonderfully. Tests still fail occasionally, but the ajax-loader won't be the cause anymore.

**Pull request merged**